frappe/core/doctype/user_permission/user_permission.js

```diff
@@ -45,7 +45,7 @@ frappe.ui.form.on('User Permission', {
 	set_applicable_for_constraint: frm => {
 		frm.toggle_reqd('applicable_for', !frm.doc.apply_to_all_doctypes);
 		if (frm.doc.apply_to_all_doctypes) {
-			frm.set_value('applicable_for', null);
+			frm.set_value('applicable_for', null, null, true);
```

Comment:
Doesn't seem like a good solution to me. When does it happen?

Reply:
On refresh, if apply_to_all_doctypes is set applicable_for is changing to null which is triggering dirty() and changing the status to "Not Saved", to avoid that Suraj and I discussed setting a flag to avoid triggering dirty.